### ISSUE TITLE
Internal error when using filter keywords

### ISSUE THREAD

**Issue by davebaldwin** [olifolkerd/tabulator#3019]
Uncaught TypeError: filterVal.toLowerCase is not a function
    keywords tabulator.js:13912
    filterFunc tabulator.js:13602
    filterRecurse tabulator.js:13859
    filterRow tabulator.js:13834
    filterRow tabulator.js:13833
    filter tabulator.js:13810
    filter tabulator.js:13809
    refreshActiveData tabulator.js:3009
    filterRefresh tabulator.js:2931
    setFilter tabulator.js:7178

**Describe the bug**
Using keywords filtering given this internal error on Firefox but also Safari

**Tabulator Info**
4.8.2

**Working Example**
No filtering takes place and looking in debug console of browser shows the error as noted above.  Don't really see that JS Fiddle version will show anything.

YOU MUST include a link to a JS Fiddle or Codepen that demonstrates the problem, it is very hard to diagnose an issue from a simple description.

**To Reproduce**
A step by step guide to recreate the issue in your JS Fiddle or Codepen:

**Expected behavior**
Table to be filtered based on keywords criteria

**Screenshots**

**Desktop (please complete the following information):**
OSX 10.15.6
 Firefox 80.0.1
Safari 14.0

**Additional context**

**Comment by davebaldwin** [comment]
Apologies but this is with 4.7.2 and not 4.8.2

**Comment by olifolkerd** [comment]
Have you tested with the latest version. If it works on the latest version please close this issue.

Cheers

Oli

**Comment by davebaldwin** [comment]
Hi Oli

I though I was using the latest version, but when I found out I wasn’t I loaded it, but now I am getting 
TypeError: Object is not a constructor (evaluating 'new Tabulator’)

So I am trying to resolve this, but with out much luck so far.

Dave.

**Comment by olifolkerd** [comment]
Hey @davebaldwin 

Can you include a link to a JS fiddle or codepen that replicates your issue, it is pretty much impossible to get to the bottom of the issue from just an error log, we need to understand how your table is being setup and used in order to help you out.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @davebaldwin 

Looking into this it is because you are passing a non string value into the filter. and the filter is only intended to filter strings.

Cheers

Oli :)

**Comment by davebaldwin** [comment]
Hi Oli,

Sorry I never got back to you on this.  I am pretty sure I was passing in a string.  I changed over to using  regexp for the filter operation and never looked at it closer.

I was never able to get the latest version working.  Nothing had changed from my end other than using yarn to update tabulator.  The problem I was having looked similar to a bug someone else reported a while ago on an earlier version that got fixed.  I have been under some time pressure to make progress so as I didn’t need any of the new features I just regressed to the version I knew worked for me.

Great library, by the way.  Really clean and well thought out.

Dave.
